viewing_party/main.py

```diff
@@ -0,0 +1,271 @@
+
+# create movie function
+
+def create_movie(title, genre, rating):
+    new_movie = {
+        "title": title,
+        "genre": genre, 
+        "rating": rating 
+    }
+    if title and genre and rating:
+        return new_movie
+    else:
+        return None
+    
+#adds movie to user watched
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+#adds movie to watchlist
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+#moves movie from watchlist to empty watched list   
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            add_to_watched(user_data, movie)
+    return user_data
+#Calculate the average rating of all movies in watched list
+#return the average rating
+def get_watched_avg_rating(user_data):
+    if user_data["watched"]:  
+        count = 0
+        for movie in user_data["watched"]:
+            count += movie["rating"]
+            average = count / len(user_data["watched"])
+        return average
+    else: 
+        return 0.0
+    
+#finds the most watched genre
+def get_most_watched_genre(user_data):
+    most_watched_dict = {} 
+    popular_genre = None
+    for movie in user_data["watched"]:
+        if movie["genre"] not in most_watched_dict:
+            most_watched_dict[movie["genre"]] = 1
+        else: 
+             most_watched_dict[movie["genre"]] += 1
+        popular_genre = max(most_watched_dict, key = most_watched_dict.get)
+        
+    return popular_genre
+
+# compares list1 to list2 and returns unique info from list1
+def get_unique(list1, list2):
+    unique_list = []
+    for list_element in list1:
+        is_unique = True
+        for list_item in list2:
+            if list_element['title'].lower() == list_item['title'].lower():
+                is_unique = False
+                break
+
+        if is_unique:
+            unique_list.append(list_element)
+    
+    return unique_list
+#returns list of movies in the users data absent from their friends data
+def get_unique_watched(user_data):
+    user_titles_list = []
+    friends_list = []
+    for movie in user_data["watched"]:
+        user_titles_list.append(movie)
```

Comment:
I assume you copied this so that the function doesn't have the unintended "side-effects" of updating `user_data`? If so, great job applying what we learned last week!